js/pages/characterizations/components/characterizations/characterization-view-edit/characterization-exec-wrapper.js

```diff
@@ -3,20 +3,37 @@ define([
     'text!./characterization-exec-wrapper.html',
     'components/Component',
     'utils/CommonUtils',
-    './characterization-executions',
+    'const',
+    'pages/characterizations/services/CharacterizationService',
+	'pages/characterizations/services/PermissionService',
     './characterization-results',
+    'components/analysisExecution/analysis-execution-list',
 ], function (
     ko,
     view,
     Component,
-    commonUtils
+    commonUtils,
+    consts,
+    CharacterizationService,
+    PermissionService,
 ) {
     class CharacterizationExecWrapper extends Component {
         constructor(params) {
             super();
 
             this.executionId = params.executionId;
-            this.componentParams = params;
+            const extraExecutionPermissions = ko.computed(() =>!params.designDirtyFlag().isDirty() && params.design().cohorts().length);
+            this.componentParams = {
+                tableColumns: ['Date', 'Design', 'Status', 'Duration', 'Results'],
+                runExecutionInParallel: false,
+                resultsPathPrefix: '/cc/characterizations/',
+                analysisId: params.characterizationId,
+                ExecutionService: CharacterizationService,
```

Comment:
The `CharacterizationService` is lacking a 'generate' function which then causes a failure when attempting to generate a characterization.